File: src/main/java/toast/pine/Adapters/MonsterTypeAdapter.java
package toast.pine.Adapters;

import org.bukkit.persistence.PersistentDataAdapterContext;
import org.bukkit.persistence.PersistentDataType;
import toast.pine.Monsters.MonsterType;

public class MonsterTypeAdapter implements PersistentDataType<MonsterType, MonsterType> {
    @Override
    public Class<MonsterType> getPrimitiveType() {
        return MonsterType.class;
    }

    @Override
    public Class<MonsterType> getComplexType() {
        return MonsterType.class;
    }

    @Override
    public MonsterType fromPrimitive(MonsterType complex, PersistentDataAdapterContext context) {
        return complex; // Implement this based on your requirement to retrieve MonsterType from a primitive form if needed.
    }

    @Override
    public MonsterType toPrimitive(MonsterType complex, PersistentDataAdapterContext context) {
        return complex; // Implement this based on your requirement to convert MonsterType to a primitive form if needed.
    }
}

File: src/main/java/pine/toast/toastrpg/adapters/MonsterTypeAdapter.kt
package pine.toast.toastrpg.adapters

import org.bukkit.persistence.PersistentDataAdapterContext
import org.bukkit.persistence.PersistentDataType
import pine.toast.toastrpg.monsters.MonsterType


class MonsterTypeAdapter : PersistentDataType<MonsterType, MonsterType> {
    @get:JvmName("getPrimitiveTypeMonsterType")
    private val primitiveType: Class<MonsterType>
        get() = MonsterType::class.java
    @get:JvmName("getComplexTypeMonsterType")
    private val complexType: Class<MonsterType>
        get() = MonsterType::class.java

    override fun fromPrimitive(complex: MonsterType, context: PersistentDataAdapterContext): MonsterType {
        return complex // Implement this based on your requirement to retrieve MonsterType from a primitive form if needed.
    }

    override fun toPrimitive(complex: MonsterType, context: PersistentDataAdapterContext): MonsterType {
        return complex // Implement this based on your requirement to convert MonsterType to a primitive form if needed.
    }

    override fun getPrimitiveType(): Class<MonsterType> {
        return primitiveType
    }

    override fun getComplexType(): Class<MonsterType> {
        return complexType
    }
}
